WebRTC: Data channel test for empty string messages

tidoust opened the pull request:
This test creates a Data channel between two local PeerConnection instances
and checks how empty string messages (a call to "send('')" in other words)
are processed.

The steps to follow to send a message (rightly) do not distinguish between
empty strings and other types of strings, so empty strings sent on one end
should be received on the other end, see:

 http://w3c.github.io/webrtc-pc/#dom-datachannel-send

The test currently fails in Chrome and Firefox.

Note that the behavior described above is consistent with how empty strings
are managed in WebSockets: calling "send('')" on an opened WebSocket correctly
triggers a "message" event on the receiving side.

tidoust committed:
WebRTC: Data channel test for empty string messages

This test creates a Data channel between two local PeerConnection instances
and checks how empty string messages (a call to "send('')" in other words)
are processed.

The steps to follow to send a message (rightly) do not distinguish between
empty strings and other types of strings, so empty strings sent on one end
should be received on the other end, see:

 http://w3c.github.io/webrtc-pc/#dom-datachannel-send

The test currently fails in Chrome and Firefox.

Note that the behavior described above is consistent with how empty strings
are managed in WebSockets: calling "send('')" on an opened WebSocket correctly
triggers a "message" event on the receiving side.

dontcallmedom committed:
focus test only on empty string exchange

dontcallmedom commented:
see https://github.com/tidoust/web-platform-tests/pull/1 for a proposed refocused rewrite.

dontcallmedom committed:
removing extraneous empty lines

tidoust committed:
Merge pull request #1 from dontcallmedom/webrtc-datachannel-empty

focus test only on empty string exchange

dontcallmedom commented on webrtc/datachannel-emptystring.html:
```diff
@@ -0,0 +1,116 @@
+  <script src="/common/vendor-prefix.js"
+          data-prefixed-objects=
+              '[{"ancestors":["window"], "name":"RTCPeerConnection"},
+                {"ancestors":["window"], "name":"RTCSessionDescription"},
```
RTCSessionDescription is not used, no need to prefix it

dontcallmedom commented on webrtc/datachannel-emptystring.html:
```diff
@@ -0,0 +1,116 @@
+    .then(gFirstConnection.setLocalDescription.bind(gFirstConnection))
+    .then(exchangeDescription(gSecondConnection, gFirstConnection))
+    .then(function() {
+      return gSecondConnection.createAnswer()
```
missing a trailing ";" (I know, my fault)

dontcallmedom commented on webrtc/datachannel-emptystring.html:
```diff
@@ -0,0 +1,116 @@
+  function exchangeIce(pc) {
+     return function(e) {
+       if (e.candidate) {
+          pc.addIceCandidate(new RTCIceCandidate(e.candidate));
```
new RTCIceCandidate() is extraneous (again, my fault)

tidoust committed:
Dropped extraneous code, cleaned list of interfaces that need prefixes

This commit incorporates further fixes suggested by @dontcallmedom. I
dropped RTCIceCandidate from the list of objects to prefix as well,
since it is no longer used in the code.

Successfully checked in Firefox (the test fails but does not crash).
